### DISCUSSION TITLE
ResourceWarning: unclosed <ssl.SSLSocket

### DISCUSSION THREAD

**Discussion by 0n31r0s** [jertel/elastalert2#887] [Q&A] [answered]
Hello 

We have deployed ElastAlert v2 in our prod environment. While the testing was done successfully, here in prod we experience these kind of errors:

WARNING:py.warnings:/usr/local/lib/python3.10/email/feedparser.py:123: ResourceWarning: unclosed <ssl.SSLSocket fd=5, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('10.40.140.125', 54648), raddr=('10.43.250.42', 9200)>
  self._lines.extend(lines)

Can't really understand why this is happening. Does anyone have ideas?

Much appreciated.

**Comment by jertel** [comment, marked as answer]
Moved to Discussions. Please see #11 before posting.

As for the warning, I've not seen this, and I can't understand why ElastAlert2 would have anything to do with a feed parser (typically used for RSS and Atom feeds). It mentions port 9200, which is used by Elasticsearch, but that absolutely isn't a feed source. You might need to enable full debugging to isolate other log lines around that warning, to try and establish related code.

**Reply by nsano-rururu** [reply]
@0n31r0s 

What's wrong with the warning? The process works, but are you just worried about the warning?

**Reply by nsano-rururu** [reply]
This is a warning that occurs when executing the logic of python3.10.
https://github.com/python/cpython/blob/3.10/Lib/email/feedparser.py#L123

**Reply by nsano-rururu** [reply]
https://github.com/algolia/algoliasearch-client-python/issues/419
